### PR TITLE
Trigger Docker builds on release branches

### PR DESCRIPTION
**Pull request by feedmeapples**
**What changed?**

Starts Triggering Docker builds on commits into Release branches

Related PR:
https://github.com/temporalio/docker-builds/pull/23

**Why?**

Automating releases

**How did you test it?**

Tested the code with personal repos and simulating the changes:
https://github.com/feedmeapples/gha-parent
https://github.com/feedmeapples/gha-child

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

**Review by yiminc — APPROVED** [review]

File: .github/workflows/trigger-publish.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - master
+      - release/v[0-9]+.[0-9]+.[0-9]+ # https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#patterns-to-match-branches-and-tags
```
Comment (dnr) [inline .github/workflows/trigger-publish.yml:7]: First, this doesn't match any current release branches (which are named like `release/v1.14.x`).

Second, how about just make it `release/*`?

Comment (feedmeapples) [reply .github/workflows/trigger-publish.yml:7]: good catch, yea let's just make * then

File: .github/workflows/trigger-publish.yml [hunk 2]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - master
+      - release/*
   workflow_dispatch:
```
Comment (dnr) [inline .github/workflows/trigger-publish.yml:8]: btw, why do we need workflow_dispatch here? what calls this?

**Review by dnr — APPROVED** [review]

File: .github/workflows/trigger-publish.yml (hunk 2 above)
Comment (feedmeapples) [reply .github/workflows/trigger-publish.yml]: removed

**Pull request merged**